src/ai/backend/manager/models/agent.py

```diff
@@ -325,6 +331,174 @@ class Meta:
     items = graphene.List(Agent, required=True)
 
 
+async def _query_domain_groups_by_ak(
+    db_conn: SAConnection,
+    access_key: str,
+) -> tuple[str, list[uuid.UUID]]:
+    j = sa.join(
+        keypairs,
+        users,
+        keypairs.c.user == users.c.uuid,
+    )
+    query = (
+        sa.select([keypairs, users.c.domain_name])
+        .select_from(j)
+        .where(keypairs.c.access_key == access_key)
+    )
+    result = await db_conn.execute(query)
+    row = result.first()
+    domain_name = row["domain_name"]
+    user_id = row["user"]
+
+    query = (
+        sa.select([association_groups_users.c.group_id])
+        .select_from(association_groups_users)
+        .where(association_groups_users.c.user_id == user_id)
+    )
+    result = await db_conn.execute(query)
+    groud_ids = [row["group_id"] for row in result.fetchall()]
+    return domain_name, groud_ids
+
+
+async def _append_sgroup_from_clause(
+    graph_ctx,
+    query,
```

Comment:
This can be type-annotated as `sa.sql.Select` (where `sa` is `import sqlalchemy as sa`).